src/WebView.php

```diff
@@ -160,6 +166,9 @@ public function endBody(): void
      */
     public function endPage($ajaxMode = false): void
     {
+        if ($this->getViewFile() === null) {
```

Comment:
How about `\RuntimeException('endPage() should be called in views only')`?

Reply:
Are such cases possible? If not, should be warned to call `beginPage()` before `endPage()`. In this case, `getViewFile ()` cannot be null and unnecessary to check, but can check `if (BeginPage not called) {throw ...}`.

```php
$this->beginPage();
$this->beginPage();   // nested begin page
...
$this->endPage();
$this->endPage();
```

or

```
$this->endPage(); // call without beginPage
```